packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts

```diff
@@ -73,13 +77,31 @@ export class InsuredBridgeL2Client {
         instantRelayFeePct: depositRelayedEvent.returnValues.instantRelayFeePct,
         quoteTimestamp: Number(depositRelayedEvent.returnValues.quoteTimestamp),
       };
+      depositData.depositHash = this.generateDepositHash(depositData);
+      this.deposits[depositRelayedEvent.returnValues.depositId] = depositData;
```

Comment:
Is it OK if a deposit data for some reason has a hash of `""` or should we explicitly error out on this case? Currently you allow it.

Reply:
ye, that's a good point and it should defo error out in that case. I can change it to error in the `this.generateDepositHash` method.

Reply:
yeah I think you should change that. Maybe use an `assert(hash !== "")` or something even more specific

Reply:
cool. added `if (depositHash == "" || depositHash == null) throw new Error("Bad deposit hash");`